### PR TITLE
.Net: Postgres memory store add simple constructor

### PR DESCRIPTION
**Pull request by JadynWong**
### Motivation and Context

The Postgres memory store adds a simple constructor using a connection string. The `NpgsqlDataSource` is managed internally by the postgres memory store.

### Description

- Similar to the recent changes to the redis memory store. which allows the user to pass the connection string directly. the `NpgsqlDataSource` is managed internally.
- Fixed postgres integration test (caused by #2419)

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Review by lemillermicrosoft — APPROVED** [review]

File: dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresMemoryStore.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
-﻿// Copyright (c) Microsoft. All rights reserved.
+﻿
```
Comment (dmytrostruk) [inline dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresMemoryStore.cs:1]: Empty line:
```suggestion
```

File: dotnet/src/IntegrationTests/Connectors/Memory/Postgres/PostgresMemoryStoreTests.cs [hunk 2]
```diff
@@ -231,7 +231,7 @@ public async Task UpsertReplacesExistingRecordWithSameIdAsync()
         Assert.NotNull(actual);
         Assert.Equal(testRecord.Metadata.Id, key);
         Assert.Equal(testRecord2.Metadata.Id, actual.Key);
-        Assert.True(testRecord.Embedding.Span.SequenceEqual(actual.Embedding.Span));
+        Assert.False(testRecord.Embedding.Span.SequenceEqual(actual.Embedding.Span));
```
Comment (dmytrostruk) [inline dotnet/src/IntegrationTests/Connectors/Memory/Postgres/PostgresMemoryStoreTests.cs:234]: It's interesting that previously we checked that `testRecord` embeddings should be equal to `actual`. But it looks like they should be different based on test scenario, so this change is valid.

**Review by dmytrostruk — APPROVED** [review]
LGTM! Functionality verified with integration tests.

File: dotnet/src/IntegrationTests/Connectors/Memory/Postgres/PostgresMemoryStoreTests.cs (hunk 2 above)
Comment (JadynWong) [reply dotnet/src/IntegrationTests/Connectors/Memory/Postgres/PostgresMemoryStoreTests.cs:234]: Yes, it originally verified that the embeddings value had been updated after the records with the same key were updated. However, when the embedded types are updated, the is updated to verity equal.

**Review by lemillermicrosoft — APPROVED** [review]

**Pull request merged**